README.md

```diff
@@ -121,6 +121,37 @@ Each doctest has access to variables in its scope chain.
   - [Grunt](http://gruntjs.com/):
       [paolodm/grunt-doctest](https://github.com/paolodm/grunt-doctest)
 
+### Gotchas
+
+Result lines are obligatory. In the file you are going to test, a line is
+**obligatory** even if the output you expect is empty.
```

Comment:
Are you sure about this? I just tested the following file:

```javascript
// > 2 + 2
console.log ('hello');
```

Occasionally I find it useful to introduce a variable, which does not necessitate an output line. For example:

```javascript
// > let x = 2
// > x + x
// 4
```

See <https://github.com/sanctuary-js/sanctuary/blob/v2.0.0/index.js#L1749-L1760> for a *realistic* example. ;)

Could you share a small example that demonstrates the failure case, @kocielnik?

Reply:
Thanks for taking a look at this, @davidchambers!

I have just run a number of tests in DocTest. They had no empty comment line below, and in most cases they passed, just like you demonstrated. 

I attempted to pin-point the circumstances where I encounter the errors mentioned.

I noticed that DocTest-ing the following file gives me an error:

```javascript
"use strict";

// > one()
function one() {
    return 1
}

// myString :: String
//
// > myString()
// "Hello world!"
function myString() {
    return 'Hello world!'
}
```

The error goes `error: Line 14: Unexpected token :`

Adding an empty line after line 3 helps and then the test is run and failed. That is probably what I understood as the "need for a comment line after a test case". However, removing lines 8-9 helps as well, and then the test is run and passed.

Now I would rather say that the next comment line after a test case is treated as the expected result, even if it resides multiple non-commented lines below the test case. Would that be more accurate?

Reply:
Rather than document this pitfall, we could update the behaviour to match your expectations. It seems you do not expect `myString :: String` to be treated as output because it does not appear immediately after `> one()`. We could compare line numbers to decide whether a particular comment line should be treated as output.

I'm also interested in your use case. I don't see a reason to include `> one()` but not the corresponding output. Could you share a snippet of your actual code?

Reply:
In the original scenario, I was designing a test for a function to return the current timestamp.  and wanted to preview first, what the format of such a timestamp actually was.

```javascript
// Current UNIX timestamp
// 
// > typeof currentTimestamp()
// 'number'
//
// currentTimestamp :: IO () -> Number
function currentTimestamp() {

    let timestamp = +new Date();

    return timestamp;
}
```

My idea was to write a sample test that expects nothing, run it, and receive as a result a message like `FAIL: expected undefined on line 6 (got 1571984872191)`. Then I would know the format of the timestamp to expect in the future. This way, it was not a "unit test" by itself, but rather a tiny hack that I supposed could work.

> We could compare line numbers to decide whether a particular comment line should be treated as output.

I think this approach would provide exactly the precise output I hoped to get in the situation:)